src/Uno.UI.Runtime.Skia.Wpf/Rendering/SoftwareWpfRenderer.cs

```diff
@@ -50,6 +50,11 @@ public void Render(DrawingContext drawingContext)
 
 		using var _ = _fpsHelper.BeginFrame();
 
+		while (_host.RootElement is { } rootElement && (rootElement.IsArrangeDirtyOrArrangeDirtyPath || rootElement.IsMeasureDirtyOrMeasureDirtyPath))
+		{
+			rootElement.UpdateLayout();
```

Comment:
This also feels like what leads to the render being called before any layout is done could be another bug. Aren't we supposed to invalidate the layout on invalidatearrange/invalidatemeasure?

Reply:
I think you meant "invalidate the render on invalidatearrange/invalidatemeasure"? No, that shouldn't be the case. Layouting should change properties on the Visual associated with the UIElement and _that_ should invalidate the visual. In the case of TextBox/TextBlock, measuring (which called InlinesCollection.Measure) does not invalidate the render. Now, the obvious fix would be to add an invalidation line somewhere there but there is still a big problem. What's happening right now means that we're rendering partial (i.e. not fully laid out) frames. I think we should actually have both fixes.